app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,47 @@
+class CommentsController < ApplicationController
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = parent.comments.new(comment_params)
+    @comment.update_attributes(user: current_user)
+
+    if @comment.save
+      respond_to do |format|
+        format.js do
+          if parent_string == "answer"
+            render partial: 'add_new_ajax_to_a' and return
+          else
+            render partial: 'add_new_ajax_to_q' and return
+          end
+        end
+        format.any do
+        end
+      end
+      redirect_to question_path(params[:question_id])
```

Comment:
You remove the need to do the `and return` bits above if you just move this redirect up into the `format.any` block.